File: recursions/find_factorial/find_factorial.js
// Find factorial of any number (recursive and iterative)

function findFactorialRecursive(num) {
    if (num === 2) {
        return 2;
    } else if (num === 1 || num === 0) {
        return 1;
    }

    return num * findFactorialRecursive(num-1);
}

const recursive = findFactorialRecursive(4);
console.log("Recursive function: " +recursive);

function findFactorialIterative(num) {
    let counter = 0;
    let answer = 1;
    if (num === 0) {
        return answer;
    }

    for (let i = 0; i < num; ++i) {
        counter++;
        answer = answer * counter;
    }

    return answer;
}

const factorial = findFactorialIterative(5);
console.log("Iterative function: " + factorial);

File: recursions/find_factorial/find_factorial.lua
local function findFactorialRecursive(num)
    if num == 0 or num == 1 then
        return 1
    elseif num == 2 then
        return 2
    end

    return num * findFactorialRecursive(num-1)
end

local recursive = findFactorialRecursive(4)
print("Recursive functions: " .. recursive)

local function findFactorialIterative(num)
    local answer = 1

    if num == 0 then
        return answer
    end

    for i= 1, num do
        answer = answer * i
    end

    return answer
end

local iterative = findFactorialIterative(5)
print("Iterative functions: " .. iterative)
